File: clinic/src/clinical-records/dto/create-clinical-record.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import {
  IsDateString,
  IsInt,
  IsNotEmpty,
  IsString,
  MaxLength,
  IsOptional,
  Min,
} from 'class-validator';

export class CreateClinicalRecordDto {
  @ApiProperty({
    description: 'ID del paciente',
    example: 1,
    minimum: 1,
  })
  @IsInt({ message: 'El ID del paciente debe ser un número entero' })
  @Min(1, { message: 'El ID del paciente debe ser mayor a 0' })
  patientId: number;

  @ApiProperty({
    description: 'ID del tipo de tumor',
    example: 3,
    minimum: 1,
  })
  @IsInt({ message: 'El ID del tipo de tumor debe ser un número entero' })
  @Min(1, { message: 'El ID del tipo de tumor debe ser mayor a 0' })
  tumorTypeId: number;

  @ApiProperty({
    description: 'Fecha del diagnóstico (formato ISO 8601)',
    example: '2023-12-10',
    type: String,
  })
  @IsDateString(
    {},
    { message: 'La fecha debe estar en formato válido (YYYY-MM-DD)' },
  )
  diagnosisDate: string;

  @ApiProperty({
    description: 'Diagnóstico médico',
    example: 'Carcinoma ductal infiltrante',
    maxLength: 255,
  })
  @IsString()
  @IsNotEmpty({ message: 'El diagnóstico es obligatorio' })
  @MaxLength(255, { message: 'El diagnóstico no puede exceder 255 caracteres' })
  diagnosis: string;

  @ApiProperty({
    description: 'Estadio del cáncer según clasificación TNM',
    example: 'IIB',
    maxLength: 20,
  })
  @IsString()
  @IsNotEmpty({ message: 'El estadio es obligatorio' })
  @MaxLength(20, { message: 'El estadio no puede exceder 20 caracteres' })
  stage: string;

  @ApiProperty({
    description: 'Protocolo de tratamiento aplicado',
    example:
      'Quimioterapia AC-T (4 ciclos Adriamicina + Ciclofosfamida, 4 ciclos Taxol) + Radioterapia',
    maxLength: 255,
  })
  @IsString()
  @IsNotEmpty({ message: 'El protocolo de tratamiento es obligatorio' })
  @MaxLength(255, {
    message: 'El protocolo de tratamiento no puede exceder 255 caracteres',
  })
  treatmentProtocol: string;

  @ApiProperty({
    description: 'Notas sobre la evolución del paciente',
    example:
      'Paciente tolera bien el tratamiento. Sin efectos adversos graves.',
    required: false,
  })
  @IsOptional()
  @IsString()
  evolutionNotes?: string;
}

File: auth-gateway/src/main/java/com/genosentinel/auth_gateway/dto/nestjs/CreateClinicalRecordDto.java
package com.genosentinel.auth_gateway.dto.nestjs;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Datos para crear una historia clínica")
public class CreateClinicalRecordDto {

    @NotNull(message = "El ID del paciente es obligatorio")
    @Positive(message = "El ID del paciente debe ser positivo")
    @JsonProperty("patientId")
    @Schema(description = "ID del paciente", example = "1")
    private Integer patientId;

    @NotNull(message = "El ID del tipo de tumor es obligatorio")
    @Positive(message = "El ID del tipo de tumor debe ser positivo")
    @JsonProperty("tumorTypeId")
    @Schema(description = "ID del tipo de tumor", example = "1")
    private Integer tumorTypeId;

    @NotNull(message = "La fecha de diagnóstico es obligatoria")
    @Schema(description = "Fecha del diagnóstico en formato AAAA-MM-DD", example = "2024-06-15")
    private String diagnosisDate;

    @NotBlank(message = "El diagnóstico es obligatorio")
    @Schema(description = "Diagnóstico médico", example = "Melanoma en estadio I")
    private String diagnosis;

    @Schema(description = "Estado del cancer ", example = "IIB")
    private String stage;

    @Schema(description = "Protocola de tratamiento", example = "Quimioterapia AC-T (4 ciclos Adriamicina + Ciclofosfamida, 4 ciclos Taxol) + Radioterapia")
    private String treatmentProtocol;

    @Schema(description = "Notas de evolución", example = "El paciente ha respondido bien al tratamiento inicial.")
    private String evolutionNotes;
}